infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/receive/LogReplicationMetadataManager.java

```diff
@@ -490,6 +498,13 @@ public void updateSnapshotSyncStatusCompleted(String clusterId, long remainingEn
                 txn.putRecord(replicationStatusTable, key, current, null);
                 txn.commit();
 
+                snapshotSyncTimerSample
+                        .flatMap(sample -> MeterRegistryProvider.getInstance()
+                                .map(registry -> {
+                                    Timer timer = registry.timer("logreplication.snapshot.duration");
+                                    return sample.stop(timer);
```

Comment:
shouldnt this timer be stopped or cancelled if we receive a 'cancel' or 'new snapshot request' event?

Reply:
Currently I thought after get the timer's sample, the only exposed API we can access is `stop()`, which stops and records the duration. The metrics will also automatically record the count of stopped (recorded) timers, so cancelled events won't have effect on metrics results. But yeah it's more reasonable if we could cancel or stop timer under such cases.